src/OpenDebugAD7/Tracepoint.cs

```diff
@@ -0,0 +1,523 @@
+        /// </summary>
+        /// <param name="input">The logMessage to parse</param>
+        /// <returns>The new string after it has been parsed, it will check for excaped curl braces.</returns>
+        private string Parse(string input)
+        {
+            StringBuilder replace = new StringBuilder();
+            bool isEscaped = false;
+            for (int index = 0; index < input.Length; index++)
+            {
+                char c = input[index];
+
+                if (isEscaped)
+                {
+                    if (c != '{' && c != '}')
+                    {
+                        replace.Append('\\');
+                    }
+
+                    // Not interpolating, output character for user's string.
+                    replace.Append(c);
+                    isEscaped = false;
+                }
+                else
+                {
+                    if (c == '\\')
+                    {
+                        isEscaped = true;
+                    }
+                    else if (c == '$')
+                    {
+                        if (FindToken(input.Substring(index), out int endIndex))
+                        {
+                            string buffer = input.Substring(index, endIndex + 1);
+
+                            m_indexToExpressions[replace.Length] = buffer;
+
+                            replace.Append(input.Substring(index, endIndex + 1));
+
+                            index += endIndex;
+                        }
+                        else
+                        {
+                            replace.Append(c);
+                        }
+                    }
+                    else if (c == '{')
+                    {
+                        if (FindInterpolatedExpression(input.Substring(index), out int endIndex))
+                        {
+                            string buffer = input.Substring(index, endIndex + 1);
+
+                            if (!string.IsNullOrWhiteSpace(buffer))
+                            {
+                                m_indexToExpressions[replace.Length] = buffer;
+                            }
+
+                            replace.Append(input.Substring(index, endIndex + 1));
+
+                            index += endIndex;
+                        }
+                        else
+                        {
+                            throw new InvalidTracepointException();
+                        }
+                    }
+                    else if (c == '"')
+                    {
+                        if (FindEndQuote(input.Substring(index), out int length))
+                        {
+                            length = length + 1;
+                        }
+                        else
+                        {
+                            length = input.Length - index;
+                        }
+                        replace.Append(input.Substring(index, length));
+                        index += length - 1;
```

Comment:
Does VS special case quoted strings also? This feels like it isn't really necessary -- if I evaluated `"{myVarWithValue1}"`, getting my `"1"` seems okay to me. But if VS does it also, I don't have a problem doing the same.